### ISSUE TITLE
Refresh cached dns periodically

### ISSUE THREAD

**Issue by DennisPan** [brightcove/hot-shots#104]
Currently, enabling `cacheDns` caches the address indefinitely. If the ip changes, then the messages are dropped. Fortunately, in UDP's case, `socket.send` does not throw if the ip is no longer there.

Wondering if this has been brought up before? If we implement something that does a new `dns.lookup` (e.g. every 30 seconds, or make it configurable), would that be a good solution? Happy to submit a PR for that.

**Comment by bdeitte** [comment]
I haven't heard this brought up before, but it definitely makes sense that this could be an issue.  I think this should be configurable, and I think a higher-but-not-indefinite lookup would be good, perhaps 5 minutes?  Thanks for any PR on it!

**Comment by msiebuhr** [comment]
I have a sort-of-similar problem where data sent to a Datadog-agent doesn't arrive after is has been re-scheduled (... inside containers ... inside Kubernetes; full story on [stack overflow](https://stackoverflow.com/questions/58094024/intra-host-udp-traffic-missing-after-destination-pod-re-created).

I considered doing a hotfix to force re-creation of the underlying socket every 30s or so:

```javascript
const statsd = new hotShots.Client(...);
setInterval(() => statsd.close(), 30 * 1000);
```

**Comment by msiebuhr** [comment]
> I have a sort-of-similar problem where data sent to a Datadog-agent doesn't arrive after is has been re-scheduled (... inside containers ... inside Kubernetes; full story on [stack overflow](https://stackoverflow.com/questions/58094024/intra-host-udp-traffic-missing-after-destination-pod-re-created).
> 
> I considered doing a hotfix to force re-creation of the underlying socket every 30s or so:
> 
> ```js
> const statsd = new hotShots.Client(...);
> setInterval(() => statsd.close(), 30 * 1000);
> ```

Tried it out -- but as there's no way to re-connect (besides creating a whole new client), this doesn't work out terribly well.

**Comment by marciopd** [comment]
Hello guys, I'm having this issue recurrently in a Kubernetes cluster.
I've created a [pull request](https://github.com/brightcove/hot-shots/pull/158) as a proposal to fix it.

The dns ttl is set as `60s`per default and can be configured through option `dnsTtlMilliseconds `.
If we want to keep backwards compatibility, of course we could change this.

I would appreciate some comments/suggestions.

**Comment by bdeitte** [comment]
Released as option cacheDnsTtl in v7.2.0
